orderedmap_test.go

```diff
@@ -263,6 +263,19 @@ func TestMove(t *testing.T) {
 	assert.NotEqual(t, err, nil)
 }
 
+func TestNewEdgeCases(t *testing.T) {
+	zero := New[int, string](0)
+	assert.Empty(t, zero.Len())
+
+	assert.PanicsWithValue(t, "too many arguments to New[K,V]()", func() {
+		_ = New[int, string](1, 2, 3)
+	})
+
+	om := New[int, string](-1)
```

Comment:
I'm actually surprised that this doesn't panic...? That goes straight to `make(map[K]*Pair[K, V], capacity[0])`, and make does panic when given a < 0 capacity?

Reply:
No panic, just `min(capacity, 0)` 
See https://go.dev/play/p/sFwJi7omFA9

Reply:
Interesting. https://go.dev/play/p/TiqBeNAUqe- does panic. I'll have to look more into that one :)